perfkitbenchmarker/scripts/wait_for_command.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python2
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -*- coding: utf-8 -*-
+"""Waits for a command started by execute_command.py to complete.
+
+Blocks until a command wrapped by "execute_command.py" completes, then mimics
+the wrapped command, copying the wrapped command's stdout/stderr to this
+process' stdout/stderr, and exiting with the wrapped command's status.
+"""
+
+import fcntl
+import optparse
+import os
+import shutil
+import sys
+import threading
+
+
+def main():
+  p = optparse.OptionParser()
+  p.add_option('-o', '--stdout', dest='stdout',
+               help="""Read stdout from FILE.""", metavar='FILE')
+  p.add_option('-e', '--stderr', dest='stderr',
+               help="""Read stderr from FILE.""", metavar='FILE')
+  p.add_option('-s', '--status', dest='status', metavar='FILE',
+               help='Get process exit status from FILE. '
+               'Will block until a shared lock is acquired on FILE.')
+  p.add_option('-d', '--delete', dest='delete', action='store_true',
+               help='Delete stdout, stderr, and status files when finished.')
+  options, args = p.parse_args()
+  if args:
+    print >> sys.stderr, 'Unexpected arguments: {0}'.format(args)
+    return 1
+
+  missing = []
+  for option in ('stdout', 'stderr', 'status'):
+    if getattr(options, option) is None:
+      missing.append(option)
+
+  if missing:
+    p.print_usage()
+    msg = 'Missing required flag(s): {0}'.format(
+        ', '.join('--' + i for i in missing))
+    print >> sys.stderr, msg
+    return 1
+
+  with open(options.stdout, 'r') as stdout:
+    with open(options.stderr, 'r') as stderr:
+      with open(options.status, 'r') as status:
+        fcntl.lockf(status, fcntl.LOCK_SH)
+        return_code = int(status.read())
+        status.close()
+
```

Comment:
Would be useful at here to log how large the stderr file to be copied is, so people would have an expectation on how soon it would finish.